File: An2/Sem1/MAP/MAP-Java/MAP-Lab7/src/Controller/Validator.java
package Controller;

import Model.Student;
import Repository.Repository;
import Repository.Stack;
import Utils.MyException;

/**
 * Created with IntelliJ IDEA.
 * User: mihai
 * Date: 11/28/13
 * Time: 3:42 PM
 * To change this template use File | Settings | File Templates.
 */
public class Validator {
    private Repository repo;

    public Validator(Repository repo){
        this.repo = repo;
    }

    private boolean studentExists(int id) throws MyException {

        Stack<Student> students = repo.getAllElements();
        while (!students.isEmpty()){
            Student currStudent;
            currStudent = students.pop();
            if (currStudent.getID() == id)
                return true;
        }
        return false;
    }

    private boolean nameIsEmpty(String name){
        if (name.length() == 0) return true;
        return false;
    }

    public void validateStudent(Student s) throws MyException{
        if (this.nameIsEmpty(s.getName()))
            throw new MyException("Name cannot be empty!");
        if (this.studentExists(s.getID()))
            throw new MyException("Student id already exists!");
    }

}

File: An2/Sem1/MAP/MAP-C#/MAP-Lab6/MAP-Lab6/Controller/Validator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using MAP_Lab6.Repository;
using MAP_Lab6.Model;
using MAP_Lab6.Utils;

namespace MAP_Lab6.Controller
{
    class Validator
    {
        private StudentRepository<Student> repo;

        public Validator(StudentRepository<Student> repo)
        {
            this.repo = repo;
        }

        private bool studentExists(int id)
        {
            MAP_Lab6.Repository.Stack<Student> students = repo.getAllElements();
            while (!students.isEmpty())
            {
                Student currStudent;
                currStudent = students.pop();
                if (currStudent.getID() == id)
                    return true;
            }
            return false;
        }

        private bool nameIsEmpty(String name)
        {
            if (name.Length < 1) return true;
            else return false;
        }

        public void validateStudent(Student s)
        {
            if (this.nameIsEmpty(s.getName()))
                throw new MyException("Name cannot be empty!\n");
            if (this.studentExists(s.getID()))
                throw new MyException("Student id already exists!\n");
        }
    }
}
